packages/byods/src/token-manager/index.ts

```diff
@@ -70,11 +79,45 @@ export default class TokenManager {
    * const authorization = await tokenManager.getOrgServiceAppAuthorization('org-id');
    */
   public async getOrgServiceAppAuthorization(orgId: string): Promise<OrgServiceAppAuthorization> {
-    if (!this.serviceAppAuthorizations[orgId]) {
-      return Promise.reject(new Error('Service app authorization not found'));
+    let token = await this.getTokenFromAdapter(orgId);
```

Comment:
@bhabalan , changed this implementation. Now I abstract the getToken part into a private method, hence whenever it throws an error (ie adapter logic), it will be visible from the new getToken private method. Wanted to keep adapter logic localised to one function only.
In this new function, we simply extract the token, perform the validation and refresh, if required.